build/lib/builder.js

```diff
@@ -37,16 +37,27 @@ function determineBabelOptions(babelOptions) {
 	const options = {
 		...babelOptions,
 		useBuiltIns: 'entry',
-		corejs: 3
+		corejs: '3.7.0' // this should match the dependency version exactly
 	};
 	// extract out options.transform used by babel-plugin-transform-titanium
 	const transform = options.transform || {};
 	delete options.transform;
 
+	// Validate that the core-js dependency and version are matched, we use require here to make
+	// use of the cache if this is called multiple times
+	const packageLock = require('../../package-lock');
```

Comment:
So I think we should just read the package-lock and use the value. We're already reading the package-lock to check they're in sync - but we're not failing the build if they aren't, just warning. It's better than nothing, but if we can automatically enforce they're in sync we should; just do it. Otherwise I suspect they'll get out of sync and the warning will get lost in our mountains of output and no one will notice.

Reply:
Updated!